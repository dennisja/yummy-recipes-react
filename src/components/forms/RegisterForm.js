import React, { Component } from 'react';
import Configs from '../../configs/Configs';
import User from '../../helpers/User';
import Token from '../../helpers/Token';

class RegisterForm extends Component {
    static initialState = {
        firstname: '',
        lastname: '',
        email: '',
        password: '',
        c_password: '',
    }

    state = RegisterForm.initialState;

    handleInputChange = (event) => {
        const target = event.target;
        this.setState({
            [target.name]: target.value
        });
    }

    handleSubmit = (event) =>{
        event.preventDefault();
        const {baseUrl, registerUrl} = Configs.api;

        User.registerUser(this.state)
        .then((response)=>{
            alert('yeah')
            console.log(response)
            this.setState(RegisterForm.initialState);
        })
        .catch((error)=>{
            const { data, status } = error.response;
            console.log(data)
            alert(data.errors)
        })
    }

    render() {
        return (
            <form onSubmit={this.handleSubmit}>
                <div className="input-field">
                    <label htmlFor="firstname">First Name:</label>
                    <input type="text" name="firstname" id="firstname" onChange={this.handleInputChange} value={this.state.firstname} required/>
                </div>
                <div className="input-field">
                    <label htmlFor="lastname">Last Name:</label>
                    <input type="text" name="lastname" id="lastname" onChange={this.handleInputChange} value={this.state.lastname} required/>
                </div>
                <div className="input-field">
                    <label htmlFor="email">Email:</label>
                    <input type="email" name="email" id="reg_email" onChange={this.handleInputChange} value={this.state.email} required/>
                </div>
                <div className="input-field">
                    <label htmlFor="password">Password:</label>
                    <input type="password" name="password" id="reg_password" onChange={this.handleInputChange} value={this.state.password} required/>
                </div>
                <div className="input-field">
                    <label htmlFor="c_password">Confirm Password: </label>
                    <input type="password" name="c_password" id="c_password" onChange={this.handleInputChange} value={this.state.c_password} required/>
                </div>
                <div className="center-align">
                    <input type="submit" className="btn btn-small orange" name="register" id="register" value="Register" />
                </div>
            </form>
        );
    }
}

export default RegisterForm;
