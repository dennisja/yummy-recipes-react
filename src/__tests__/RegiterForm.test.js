import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import RegisterForm from '../components/forms/RegisterForm';

describe('Test RegisterForm', () => {
    const formWrapper = shallow(<RegisterForm />);
    const preventDefault = jest.fn();


    it('renders properly', () => {
        expect(shallowToJson(formWrapper)).toMatchSnapshot();
    });

    it('renders a form element', () => {
        expect(formWrapper.find('form')).toHaveLength(1);
    });

    it('renders the correct form fields', () => {
        expect(formWrapper.find('#firstname').length).toBe(1);
        expect(formWrapper.find('#lastname').length).toBe(1);
        expect(formWrapper.find('#reg_email').length).toBe(1);
        expect(formWrapper.find('#reg_password').length).toBe(1);
        expect(formWrapper.find('#c_password').length).toBe(1);
        expect(formWrapper.find('#register').length).toBe(1);
    });

    it('submits data', () => {
        formWrapper.find('form').simulate('submit', { preventDefault });
        expect(preventDefault).toBeCalled();
    });
});
