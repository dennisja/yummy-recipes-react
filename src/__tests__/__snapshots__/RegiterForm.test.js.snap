// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RegisterForm renders properly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="input-field"
  >
    <label
      htmlFor="firstname"
    >
      First Name:
    </label>
    <input
      id="firstname"
      name="firstname"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="input-field"
  >
    <label
      htmlFor="lastname"
    >
      Last Name:
    </label>
    <input
      id="lastname"
      name="lastname"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="input-field"
  >
    <label
      htmlFor="email"
    >
      Email:
    </label>
    <input
      id="reg_email"
      name="email"
      onChange={[Function]}
      required={true}
      type="email"
      value=""
    />
  </div>
  <div
    className="input-field"
  >
    <label
      htmlFor="password"
    >
      Password:
    </label>
    <input
      id="reg_password"
      name="password"
      onChange={[Function]}
      required={true}
      type="password"
      value=""
    />
  </div>
  <div
    className="input-field"
  >
    <label
      htmlFor="c_password"
    >
      Confirm Password: 
    </label>
    <input
      id="c_password"
      name="c_password"
      onChange={[Function]}
      required={true}
      type="password"
      value=""
    />
  </div>
  <div
    className="center-align"
  >
    <input
      className="btn btn-small orange"
      id="register"
      name="register"
      type="submit"
      value="Register"
    />
  </div>
</form>
`;
